pallets/crowdloan-reward/src/tests.rs

```diff
@@ -97,49 +109,49 @@ fn setters_ok() {
     .existential_deposit(1)
     .build(|| {
       System::set_block_number(1);
-      CrowdloanReward::initialize(50, 20, 4, 3).unwrap();
+      CrowdloanReward::initialize(
+        50,
+        Perbill::from_percent(20),
+        4,
+        3
+      ).unwrap();
     })
     .execute_with(|| {
       assert!( CrowdloanReward::set_vesting_start(Origin::signed(1), 1).is_ok() );
       assert!( CrowdloanReward::set_vesting_period(Origin::signed(1), 55555).is_ok() );
       assert!( CrowdloanReward::set_direct_payout_ratio(Origin::signed(1), 9).is_ok() );
+      assert!( CrowdloanReward::set_conversion_rate(Origin::signed(1), 80).is_ok() );
+      assert!( CrowdloanReward::set_vesting_start(Origin::root(), 1).is_ok() );
+      assert!( CrowdloanReward::set_vesting_period(Origin::root(), 55555).is_ok() );
+      assert!( CrowdloanReward::set_direct_payout_ratio(Origin::root(), 9).is_ok() );
       assert!( CrowdloanReward::set_conversion_rate(Origin::root(), 80).is_ok() );
     });
 }
 
-
-#[test]
-fn elapsed_time() {
-  TestExternalitiesBuilder::default()
-    .existential_deposit(1)
-    .build(|| System::set_block_number(3) )
-    .execute_with(|| {
-      assert_noop!(
-        CrowdloanReward::set_vesting_start(Origin::root(), 2),
-        CrowdloanRewardError::<MockRuntime>::ElapsedTime
-      );
-    });
-}
-
 #[test]
 fn reward_participant() {
```

Comment:
Can we create a test that verifies 0% upfront reward?

Reply:
Yes. Will be done.

Reply:
let's add this later in ur next pr